zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/impl/filters/FilterRepository.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.impl.filters;
+
+import io.camunda.zeebe.gateway.Loggers;
+import io.camunda.zeebe.gateway.impl.configuration.FilterCfg;
+import io.camunda.zeebe.util.ReflectUtil;
+import io.camunda.zeebe.util.jar.ExternalJarLoadException;
+import io.camunda.zeebe.util.jar.ExternalJarRepository;
+import jakarta.servlet.Filter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.agrona.LangUtil;
+import org.slf4j.Logger;
+
+/** Loads and holds references to the configured filters. */
```

Comment:
💭 We could add a hint to fix the gRCP-related class when we fix something here and vice versa.

Reply:
Yup, good idea. I should have added it once I decided it was too much work to re-use the common logic.